src/win/wjoydxnu.cpp

```diff
@@ -550,6 +557,38 @@ static BOOL CALLBACK object_enum_callback(LPCDIDEVICEOBJECTINSTANCE lpddoi, LPVO
       }
    }
 
+   // The first two axis (not X or Y) are stored and used as the secondary stick.
+   // The order of the enumeration is assumed to give us the X followed by the Y axis. However,
+   // an override is provided in the case that the second axis seen here is for Z, in which case that
+   // will be used as the X axis. This is to fit behaviors seen on specific controllers:
+   // - PS4 DualShock
+   // - Stadia
+   // ...it may be that using `Z` really is common for some input devices.
+   //
+   // Solution came from https://www.gamedev.net/forums/topic/613913-directinput-identifying-second-thumbstick/
+   if (j)
```

Comment:
Now that we have a backwards compatibility mechanism, we can merge a version of this. Could you add check like this https://github.com/liballeg/allegro5/blob/8575e373ab79c7bc25f8425990fc6780e75e69c8/src/macosx/hidjoy.m#L192C4-L192C76 here, so this behavior can be configured?

Also, could `j` be called something more informative, like... `offset_override` maybe?